S2VX.Game/Story/Note/HoldNote.cs

```diff
@@ -22,20 +23,64 @@ public abstract class HoldNote : S2VXNote {
         [Resolved]
         private S2VXStory Story { get; set; }
 
-        [BackgroundDependencyLoader]
-        private void Load() => AddInternal(SliderPath);
-
-        public Path SliderPath { get; set; } = new() {
+        public SliderPath SliderPath { get; set; } = new() {
             // Show slider path behind note
             Depth = 100,
             Anchor = Anchor.Centre
         };
 
+        [BackgroundDependencyLoader]
+        private void Load() => AddInternal(SliderPath);
+
         protected override void UpdatePosition() {
             UpdateIndicator();
             UpdateSliderPath();
         }
 
+        // These are static helpers for calculating hold related values
+        // TODO: There's probably a better way to share this between HoldApproach and HoldNote for calculating these values
+        // TODO: Optimize these so that they are called once on initialization instead of ran every update loop
+        private static IEnumerable<Vector2> CombineAllCoordinates(Vector2 coordinates, List<Vector2> midCoordinates, Vector2 endCoordinates)
+            => new List<Vector2> { coordinates }.Concat(midCoordinates.Append(endCoordinates));
```

Comment:
this is just a comment. lmao

Reply:
I left these in intentionally. For this change, I didn't want to spend too much time refactoring the structure of Notes and Approaches. These are suggestions to optimize some performance. We can make stories from these, but I wouldn't want to put them into any upcoming milestones since they're not really needed.

Reply:
nah was just chuckling at the way this was constructed